app/src/main/java/org/mozilla/fenix/ext/Activity.kt

```diff
@@ -29,12 +33,35 @@ fun Activity.enterToImmersiveMode() {
  * Calls [Activity.reportFullyDrawn] while also preventing crashes under some circumstances.
  */
 fun Activity.reportFullyDrawnSafe() {
+    // An exception is thrown when calling reportFullyDrawn on some Samsung devices. We could catch
+    // the exception but catching exceptions is slow so we disable the call on all Samsung devices
+    // except the ones we're using for performance testing.
+    //
+    // We were unable to identify the root cause of the Exception but suspect it's related to Samsung
+    // security features. See
+    // https://github.com/mozilla-mobile/fenix/issues/12345#issuecomment-655058864 for details.
+    //
+    // We briefly measured how slow the Exception code path is here: on a Galaxy S5, it's ~1ms, ~6x
```

Comment:
@mcomella this is the first time I've seen code trying to reduce start-up time with 1ms, I'm impressed!

Reply:
😓